### DISCUSSION TITLE
File size upload

### DISCUSSION THREAD

**Discussion by phamtruongvinh** [nextcloud/all-in-one#975] [Questions] [answered]
I use NC AIO to install and use NC.
I see in nextcloud-aio-nextcloud ENV: PHP_UPLOAD_LIMIT 10G
But I still can't upload files >100mb.
Please help me fix this.

**Comment by Zoey2936** [comment, marked as answer]
What's your output of `docker exec nextcloud-aio-nextcloud php -i | grep max`?
And do you maybe use a reverse proxy?
And how do you upload these files? Over the Web, the Nextcloud desktop client or a mobile app or ...?

**Reply by phamtruongvinh** [reply]
> What's your output of `docker exec nextcloud-aio-nextcloud php -i | grep max`? And do you maybe use a reverse proxy? And how do you upload these files? Over the Web, the Nextcloud desktop client or a mobile app or ...?

Thanks for the reply,
this is the output for your command:
```
log_errors_max_len => 1024 => 1024
max_execution_time => 0 => 0
max_file_uploads => 20 => 20
max_input_nesting_level => 64 => 64
max_input_time => -1 => -1
max_input_vars => 1000 => 1000
post_max_size => 10G => 10G
upload_max_filesize => 10G => 10G
zend.exception_string_param_max_len => 15 => 15
ldap.max_links => Unlimited => Unlimited
memcached.sess_lock_max_wait => not set => not set
memcached.sess_lock_wait_max => 150 => 150
session.gc_maxlifetime => 1440 => 1440
unserialize_max_depth => 4096 => 4096
opcache.jit_max_exit_counters => 8192 => 8192
opcache.jit_max_loop_unrolls => 8 => 8
opcache.jit_max_polymorphic_calls => 2 => 2
opcache.jit_max_recursive_calls => 2 => 2
opcache.jit_max_recursive_returns => 2 => 2
opcache.jit_max_root_traces => 1024 => 1024
opcache.jit_max_side_traces => 128 => 128
opcache.max_accelerated_files => 10000 => 10000
opcache.max_file_size => 0 => 0
opcache.max_wasted_percentage => 5 => 5
```
I using Cloudflare tunnel as reverse proxy.
And using IOS client for upload.

**Reply by szaimen** [reply]
> Nextcloud tunnel

you mean a Cloudflare tunnel?

I think it is known that Cloudflare has a limitation of 100MB for uploads....

at least as long uploads do not get chunked...

**Reply by phamtruongvinh** [reply]
Thanks, I figured out the problem.

**Reply by szaimen** [reply]
What was the solution? (May help people with the same issue in the future)

**Comment by AndreScalaPT** [comment]
Hi, what was the solution?
I have this same configuration and still have the problem also!

Thanks

**Reply by Zoey2936** [reply]
https://developers.cloudflare.com/cache/about/default-cache-behavior#customization-options-and-limitations
![grafik](https://user-images.githubusercontent.com/75573284/211169720-a501de57-82fa-492c-9212-9485cba936a4.png)

**Reply by AndreScalaPT** [reply]
You are my idol!
Do I need to change also in the config file in the server?

Edit: owww but we need to pay to increase this! But then I'll not be able to upload large videos from my phone? Should I need a difference solution away form Cloudflare? 

Thanks

**Reply by Zoey2936** [reply]
I think you can't change here much…
You could only remove Cloudflare out of your setup or buy one of the non-free Cloudflare plans, but I think the prices are very expensive for this small upload increment, so I wouldn't recommend it.

**Reply by AndreScalaPT** [reply]
I've just edited my comment about that!
So the only solution is not use Cloudflare?

The free solution I mean.

**Reply by Zoey2936** [reply]
> I've just edited my comment about that! So the only solution is not use Cloudflare?
> 
> The free solution I mean.

I think yes...

**Reply by AndreScalaPT** [reply]
Ok Zoey!
Thank for your awesome help.

Kind regards,
André
